### PR TITLE
Prepare paper for JOSS

### PR DESCRIPTION
**Pull request by oxinabox**
Author Guidelines: https://joss.readthedocs.io/en/latest/submitting.html

### CODE REVIEW

File: paper.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+ - name: The University of Western Australia
+ - name: Birla Institute of Technology and Science, Pilani
+
+date: 24 Aug 2018
+bibliography: paper.bib
+---
+
+# Summary
+
+DataDepsGenerators.jl is an tool written to help uses of the Julia programming language ()[@Julia]),
```
Comment (daschw) [inline paper.md]: users

File: paper.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+to observe best practices when making use of published datasets.
+Using the metadata present in published datasets, in generates the code for the data dependency registration blocks required by DataDeps.jl ([@2018arXiv180801091W]).
+These registration blocks are effectively executable metadata,
+which can be resolved by DataDeps.jl to download the dataset.
+They include a message that is displayed to the user whenever the data set is automatically downloaded.
+This message should include provenance information on the dataset,
+so that downstream users know its original source and details on its processing.
+
+DataDepsGenerators.jl attempt to use the metadata available for a dataset to capture and record:
```
Comment (daschw) [inline paper.md]: attempts (?)

File: paper.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+ - The creation date, publication date, and the date of the most recent modification.
+ - The license that the dataset is released under.
+ - The formatted bibliographic details of any paper about or relating to the dataset.
+ - The formatted bibliographic details of how to cite the dataset itself.
+ - A list of URLs where the files making up the dataset can be downloaded.
+ - A corresponding list of file hashes, such as MD5 or SHA256, to validate the files after download.
+ - A description of the dataset.
+
+Depending on the APIs supported by the repository some of this information may not be available.
+DataDepsGenerators.jl makes a best effort attempt to acquire as much provenance information as possible.
```
Comment (daschw) [inline paper.md]: effort to attempt

File: paper.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+Where multiple APIs are supported, it makes use of all APIs possible, merging their responses to fill any gaps.
+It thus often produces higher quality and more comprehensive dataset metadata that in available from any one source.
```
Comment (daschw) [inline paper.md]: than available ...

File: paper.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    - Most well known for hosting code; but is fairly regularly used to host versioned datasets.
+ - [CKAN](http://ckan.org/)
+    - This is the system behind a large number of government open data initiatives;
+	- such as Data.Gov](https://data.gov), [data.gov.au](https://data.gov.au/), and the [European Data Portal](https://www.europeandataportal.eu/)
+ - Embedded JSON-LD fragments in HTML pages.
+    - This is commonly used on many websites to describe their datasets.
+	- Including many of those listed above.
+	- But also [Zenodo](https://zenodo.org/), [Kaggle Datasets](https://www.kaggle.com/datasets), all [DataVerse](https://dataverse.org/) sites and many others.
+
+DataDepsGenerators.jl as the name suggests, generated static code which the user can add into their project's julia source code to make use of with DataDeps.jl.
```
Comment (daschw) [inline paper.md]: generates

**Review by daschw — COMMENTED** [review]
I have not submitted to JOSS before, so I can't help you in this respect, but I found some typos.

**Comment by oxinabox** [comment]
@daschw  thanks! I am terrible for typos.

**Pull request merged**